src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.CompeltionListUpdater.cs

```diff
@@ -27,15 +27,35 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.AsyncComplet
 {
     internal partial class ItemManager
     {
-        private sealed class CompletionListUpdater
+        private sealed class CompletionListUpdater : IDisposable
         {
-
-            public IAsyncCompletionSession Session { get; }
-            public AsyncCompletionSessionDataSnapshot Data { get; }
-            public RecentItemsManager RecentItemsManager { get; }
-            public IGlobalOptionService GlobalOptions { get; }
-            public CancellationToken CancellationToken { get; }
-            public CompletionHelper DefaultCompletionHelper { get; }
+            private IAsyncCompletionSession Session { get; }
+            private AsyncCompletionSessionDataSnapshot Data { get; }
+            private RecentItemsManager RecentItemsManager { get; }
+            private IGlobalOptionService GlobalOptions { get; }
+            private CancellationToken CancellationToken { get; }
+
+            private bool HasSuggestedItemOptions { get; }
+            private string FilterText { get; }
+            private CompletionTriggerKind InitialRoslynTriggerKind { get; }
+            private Document? Document { get; }
+            private CompletionService? CompletionService { get; }
+            private CompletionRules CompletionRules { get; }
+            private CompletionHelper CompletionHelper { get; }
+            private bool HighlightMatchingPortions { get; }
+
+            private List<MatchResult<VSCompletionItem>> ItemsToBeIncluded { get; }
+
+            private CompletionTriggerReason TriggerReason => Data.Trigger.Reason;
+            private CompletionFilterReason FilterReason => Helpers.GetFilterReason(Data.Trigger);
+
+            private Func<ImmutableArray<(RoslynCompletionItem, PatternMatch?)>, string, ImmutableArray<RoslynCompletionItem>> FilterMethod
+                => CompletionService == null
+                    ? ((itemsWithPatternMatches, text) => CompletionService.FilterItems(CompletionHelper, itemsWithPatternMatches, text))
+                    : ((itemsWithPatternMatches, text) => CompletionService.FilterItems(Document!, itemsWithPatternMatches, text));
```

Comment:
this is super bizarre.  you ask if CompletionService is null, but if so, you still dot into it to call .FilterItems on it.

Reply:
Yea, this is super confusing, since one CompletionService referes to the type whereas the other refers to the property. Fixed